proposals/devcontainer-features.md

```diff
@@ -0,0 +1,142 @@
+# Dev container features reference
+
+Dev container features provide a smooth path for customizing your container definitions. Features are self contained units of code meant to facilitate creation of custom images for application or development containers.
+
+From a practical point of view, features are folders that contain units of code with different entrypoints for different lifecycle events.
+
+Features can be defined by a `devcontainer-feature.json` file in the root folder of the feature. The file is optional for backwards compatibility but it is required for any new features being authored.
+
+Features are to be executed in sequence as defined in `devcontainer.json`.
+
+## Folder Structure 
+
+A feature is a self contained entity in a folder. A feature release would be a tar file that contains all files part of the feature.
+
++-- feature
+|    +-- feature.json
+|    +-- aquire.sh (default)
+|    +-- install.sh (default)
+|    +-- (other files)
+
+In case `devcontainer-feature.json`  does not include a reference for the lifecycle scripts the application will look for the default script names and will execute them if available.
```

Comment:
```suggestion
In case `devcontainer-feature.json`  does not include a reference for the lifecycle scripts, the application will look for the default script names and will execute them if available.
```